backend/src/main/java/ai/verta/modeldb/configuration/AppConfigBeans.java

```diff
@@ -0,0 +1,412 @@
+import ai.verta.modeldb.telemetry.TelemetryCron;
+import ai.verta.modeldb.utils.JdbiUtil;
+import ai.verta.modeldb.utils.ModelDBHibernateUtil;
+import ai.verta.modeldb.utils.ModelDBUtils;
+import ai.verta.modeldb.versioning.FileHasher;
+import ai.verta.modeldb.versioning.VersioningServiceImpl;
+import io.grpc.BindableService;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.health.v1.HealthCheckResponse;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.spring.webmvc.SpringWebMvcTelemetry;
+import io.prometheus.client.Gauge;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Optional;
+import java.util.TimerTask;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import javax.annotation.PreDestroy;
+import javax.servlet.Filter;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EnableConfigurationProperties({FileStorageProperties.class})
+public class AppConfigBeans {
+  private static final Logger LOGGER = LogManager.getLogger(AppConfigBeans.class);
+  private final AppContext appContext;
+  protected Optional<Server> server = Optional.empty();
+
+  // metric for prometheus monitoring
+  private static final Gauge up =
+      Gauge.build()
+          .name("verta_backend_up")
+          .help("Binary signal indicating that the service is up and working.")
+          .register();
+
+  public AppConfigBeans(AppContext appContext) {
+    this.appContext = appContext;
+  }
+
+  @Bean
+  public Filter webMvcTracingFilter(OpenTelemetry openTelemetry) {
+    return SpringWebMvcTelemetry.builder(openTelemetry).build().newServletFilter();
+  }
+
+  @Bean
+  public OpenTelemetry openTelemetry(Config config) {
+    return config.getOpenTelemetry();
+  }
+
+  @Bean
+  public S3Service getS3Service() throws ModelDBException, IOException {
+    String bucketName = System.getProperty(ModelDBConstants.CLOUD_BUCKET_NAME);
+    if (bucketName != null && !bucketName.isEmpty()) {
+      return new S3Service(System.getProperty(ModelDBConstants.CLOUD_BUCKET_NAME));
+    }
+    return null;
+  }
+
+  @Bean
+  public MDBConfig config() {
+    var config = MDBConfig.getInstance();
+    App.getInstance().mdbConfig = config;
+
+    // Configure spring HTTP server
+    LOGGER.info("Configuring spring HTTP traffic on port: {}", config.getSpringServer().getPort());
+    System.getProperties().put("server.port", config.getSpringServer().getPort());
+
+    return config;
+  }
+
+  @Bean
+  Executor executor(Config config) {
```

Comment:
I wonder if this should be named "grpcExecutor()" to make it clear what it's for.

Reply:
Resolved